Username-Password-Flow/src/main/java/UsernamePasswordFlow.java

```diff
@@ -8,65 +8,121 @@
-            // try to acquire token silently. This call will fail since the token cache
-            // does not have any data for the user you are trying to acquire a token for
+            // Try to acquire token silently. This will fail on the first acquireTokenUsernamePassword() call
+            // because the token cache does not have any data for the user you are trying to acquire a token for
             result = pca.acquireTokenSilently(silentParameters).join();
+            System.out.println("==acquireTokenSilently call succeeded");
         } catch (Exception ex) {
             if (ex.getCause() instanceof MsalException) {
-
+                System.out.println("==acquireTokenSilently call failed: " + ex.getCause());
                 UserNamePasswordParameters parameters =
                         UserNamePasswordParameters
-                                .builder(SCOPE, USER_NAME, USER_PASSWORD.toCharArray())
+                                .builder(scope, username, password.toCharArray())
                                 .build();
                 // Try to acquire a token via username/password. If successful, you should see
                 // the token and account information printed out to console
                 result = pca.acquireToken(parameters).join();
+                System.out.println("==username/password flow succeeded");
             } else {
                 // Handle other exceptions accordingly
                 throw ex;
             }
         }
         return result;
     }
-}
\ No newline at end of file
+
+    /**
+     * Helper function to return an account from a given set of accounts based on the given username,
+     * or return null if no accounts in the set match
+     */
+    private static IAccount getAccountByUsername(Set<IAccount> accounts, String username) {
+        for (IAccount account : accounts) {
+            if (account.username().equals(username)) {
+                return account;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Helper function unique to this sample setting. In a real application these wouldn't be so hardcoded, for example
+     * values such as username/password would come from the user, and different users may require different scopes
+     */
+    private static void setUpSampleData() throws IOException {
+        // Load properties file and set properties used throughout the sample
+        Properties properties = new Properties();
+        properties.load(new FileInputStream(Thread.currentThread().getContextClassLoader().getResource("").getPath() + "application.properties"));
+        authority = properties.getProperty("AUTHORITY");
+        scope = Collections.singleton(properties.getProperty("SCOPE"));
+        clientId = properties.getProperty("CLIENT_ID");
+        username = properties.getProperty("USER_NAME");
+        password = properties.getProperty("USER_PASSWORD");
+        //Junk account used to pre-fill cache
```

Comment:
do not get idea behind it - in samples we definitely do no want to show how to pre-fill cache - especially with 
"junk account"

Reply:
Before these changes, the sample had this:
```
// Take first account in the cache. In a production application, you would filter
// accountsInCache to get the right account for the user authenticating.
IAccount account = accountsInCache.iterator().next();
```

However, it didn't make much sense to have a comment telling the user should do in a 'real' application when it'd be pretty simple to show it, so I made the getAccountByUsername() helper function to give an example of what they could do. But, with an empty cache that function would only search through a set that was "empty" or "had the one account we're looking for", which wouldn't be a very common scenario in an application with more than one user.

So, rather than having either a unrealistic example of getting an account or having no example at all, I figured adding a junk account to the cache would show a more 'realistic' flow of getting a set of accounts from the cache and searching through something that wasn't empty.

However, I'll definitely add more info the comment above that, explaining that it's only done for the sample and wouldn't be something that happens in a 'real' application.

Reply:
i think we need to keep sample as close to real applications as possible, for me "fake account" sounds very far from real applications

Reply:
Fair enough, I just removed accounts and TokenCacheAspect.java entirely, and I guess we can demonstrate that side of things in some new sample focused on persistence. 

I also added a few more clarifying comments/printed statements about how all of the account stuff is supposed to be empty on the first run, just so users don't think they're missing something when nothing happens.

Reply:
did you push latest changes ?

Reply:
Sorry, looks like there was an auth issue with Git when I thought I pushed them yesterday. Just pushed them now.